src/global_distrib/mod_global_distrib.erl

```diff
@@ -43,15 +43,21 @@ start(Host, Opts0) ->
 stop(Host) ->
     mod_global_distrib_utils:stop(?MODULE, Host, fun stop/0).
 
--spec get_metadata(mongoose_acc:t(), Key :: term(), Default :: term()) -> Value :: term().
+-spec get_metadata(mongoose_acc:t(), Key :: term(), Default :: term()) ->
+    Value :: term() | {error, missing_gd_structure}.
 get_metadata(Acc, Key, Default) ->
-    GD = mongoose_acc:get(global_distrib, Acc),
-    maps:get(Key, GD, Default).
+    case mongoose_acc:get(global_distrib, Acc, undefined) of
+        undefined -> {error, missing_gd_structure};
+        GD -> maps:get(Key, GD, Default)
+    end.
 
--spec get_metadata(mongoose_acc:t(), Key :: term()) -> Value :: term().
+-spec get_metadata(mongoose_acc:t(), Key :: term()) ->
```

Comment:
tricky error handling.
badkey as an exception, `{error, Reason}`, high chance that we would not match on `{error,_}` in the caller code and the tuple would "walk" around our code.